plugins/modules/cloudflare_dns.py

```diff
@@ -858,6 +908,13 @@ def main():
                     and (module.params['value'] is None or module.params['value'] == ''))):
             module.fail_json(msg="For TLSA records the params cert_usage, selector, hash_type and value all need to be defined, or not at all.")
 
+    if module.params['type'] == 'CAA':
+        if not ((module.params['flag'] is not None and module.params['tag'] is not None
```

Comment:
If `type == CAA`, `flag` and `tag` can never be `None` - the `required_if` above ensures that.